src/stmt.cpp

```diff
@@ -2641,6 +2644,7 @@ void ForeachUniqueStmt::EmitCode(FunctionEmitContext *ctx) const {
     if (exprValue == nullptr || (exprType = expr->GetType()) == nullptr ||
         llvm::dyn_cast<llvm::VectorType>(exprValue->getType()) == nullptr) {
         Assert(m->errorCount > 0);
+        ctx->EndScope();
```

Comment:
It looks like this place is also missing `ctx->EndForeach()`.

Reply:
Right, indeed.

Reply:
I still don't see `ctx->EndForeach()` after last push.